### PR TITLE
Use in-class member initializers

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

Follows #2156 using in-class member initializers in the Window module.

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Window/Unix/WindowImplX11.hpp [hunk 1]
```diff
@@ -297,25 +297,25 @@ class WindowImplX11 : public WindowImpl
...
-    bool     m_useSizeHints;   ///< Is the size of the window fixed with size hints?
-    bool     m_fullscreen;     ///< Is the window in fullscreen?
-    bool     m_cursorGrabbed;  ///< Is the mouse cursor trapped?
-    bool     m_windowMapped;   ///< Has the window been mapped by the window manager?
-    Pixmap   m_iconPixmap;     ///< The current icon pixmap if in use
-    Pixmap   m_iconMaskPixmap; ///< The current icon mask pixmap if in use
-    ::Time   m_lastInputTime;  ///< Last time we received user input
+    ::Window   m_window{0};             ///< X identifier defining our window
+    ::Display* m_display{0};            ///< Pointer to the display
+    int        m_screen;                ///< Screen identifier
```
Comment (kimci86) [inline src/SFML/Window/Unix/WindowImplX11.hpp:302]: This member was set to 0 in constructors initializer lists.
```suggestion
    int        m_screen{0};             ///< Screen identifier
```

File: src/SFML/Window/iOS/EaglContext.hpp [hunk 2]
```diff
@@ -159,12 +159,12 @@ class EaglContext : public GlContext
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
-    EAGLContext* m_context;      ///< The internal context
-    GLuint       m_framebuffer;  ///< Frame buffer associated to the context
-    GLuint       m_colorbuffer;  ///< Color render buffer
-    GLuint       m_depthbuffer;  ///< Depth render buffer
-    bool         m_vsyncEnabled; ///< Vertical sync activation flag
-    Clock        m_clock;        ///< Measures the elapsed time for the fake v-sync implementation
+    EAGLContext* m_context;             ///< The internal context
```
Comment (kimci86) [inline src/SFML/Window/iOS/EaglContext.hpp:162]: Any reason this member is not initialized here?

File: src/SFML/Window/Unix/WindowImplX11.hpp (hunk 1 above)
Comment (ChrisThrasher) [reply src/SFML/Window/Unix/WindowImplX11.hpp:302]: Oh wow great catch

File: src/SFML/Window/iOS/EaglContext.hpp (hunk 2 above)
Comment (ChrisThrasher) [reply src/SFML/Window/iOS/EaglContext.hpp:162]: You can't put that initial value `nil` in this C++ header. I tried and it failed to build. Maybe we could use `nullptr` as the initial value but idk much about Objective-C++.

Comment (ChrisThrasher) [reply src/SFML/Window/iOS/EaglContext.hpp:162]: https://github.com/ChrisThrasher/SFML/actions/runs/3505531814

I tried this out and CI passes but I'm not confident enough about the differences between `nil` and `nullptr` to make this change.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by Jim-Marsden — APPROVED** [review]

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Window/Unix/JoystickImpl.hpp [hunk 3]
```diff
@@ -116,7 +116,7 @@ class JoystickImpl
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
-    int                          m_file;                 ///< File descriptor of the joystick
+    int                          m_file{-1};             ///< File descriptor of the joystick
     char                         m_mapping[ABS_MAX + 1]; ///< Axes mapping (index to axis id)
```
Comment (eXpl0it3r) [inline src/SFML/Window/Unix/JoystickImpl.hpp:120]: Is this default initialized?

File: src/SFML/Window/Unix/WindowImplX11.hpp [hunk 4]
```diff
@@ -297,25 +297,25 @@ class WindowImplX11 : public WindowImpl
...
+    ::Cursor m_lastCursor{None}; ///< Last cursor used -- this data is not owned by the window and is required to be always valid
+    bool     m_keyRepeat{true}; ///< Is the KeyRepeat feature enabled?
+    Vector2i m_previousSize{-1, -1}; ///< Previous size of the window, to find if a ConfigureNotify event is a resize event (could be a move event only)
+    bool     m_useSizeHints{false};  ///< Is the size of the window fixed with size hints?
+    bool     m_fullscreen{false};    ///< Is the window in fullscreen?
+    bool     m_cursorGrabbed{false}; ///< Is the mouse cursor trapped?
+    bool     m_windowMapped{false};  ///< Has the window been mapped by the window manager?
+    Pixmap   m_iconPixmap{0};        ///< The current icon pixmap if in use
+    Pixmap   m_iconMaskPixmap{0};    ///< The current icon mask pixmap if in use
+    ::Time   m_lastInputTime{0};     ///< Last time we received user input
```
Comment (eXpl0it3r) [inline src/SFML/Window/Unix/WindowImplX11.hpp:318]: Indentation / alignment is off, due to clang format?

File: src/SFML/Window/Win32/WindowImplWin32.hpp [hunk 5]
```diff
@@ -269,18 +269,19 @@ class WindowImplWin32 : public WindowImpl
...
-    bool     m_resizing;        //!< Is the window being resized?
-    std::uint16_t m_surrogate; //!< First half of the surrogate pair, in case we're receiving a Unicode character in two events
-    bool          m_mouseInside;   //!< Mouse is inside the window?
-    bool          m_fullscreen;    //!< Is the window fullscreen?
-    bool          m_cursorGrabbed; //!< Is the mouse cursor trapped?
+    HWND     m_handle{nullptr};     //!< Win32 handle of the window
+    LONG_PTR m_callback{0};         //!< Stores the original event callback function of the control
+    bool     m_cursorVisible{true}; //!< Is the cursor visible or hidden?
+    HCURSOR  m_lastCursor{
+        LoadCursor(nullptr, IDC_ARROW)}; //!< Last cursor used -- this data is not owned by the window and is required to be always valid
```
Comment (eXpl0it3r) [inline src/SFML/Window/Win32/WindowImplWin32.hpp:276]: clang-format again?

**Review by eXpl0it3r — APPROVED** [review]
Looks good, just some formatting and not sure about one var that wasn't previously initialized either, but maybe should be?

File: src/SFML/Window/Unix/JoystickImpl.hpp (hunk 3 above)
Comment (ChrisThrasher) [reply src/SFML/Window/Unix/JoystickImpl.hpp:120]: https://github.com/SFML/SFML/blob/75642ef7aacca6af01f3816fa21e4724a36501a5/src/SFML/Window/Unix/JoystickImpl.cpp#L440

It's initialized in the constructor body. I think we could add `{0}` to get rid of that `std::fill` call but I'm not 100% sure I'm correct about that.

File: src/SFML/Window/Unix/WindowImplX11.hpp (hunk 4 above)
Comment (ChrisThrasher) [reply src/SFML/Window/Unix/WindowImplX11.hpp:318]: Must be so if the formatting CI jobs pass ¯\_(ツ)_/¯

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by eXpl0it3r — APPROVED** [review]

**Pull request merged**